compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -912,6 +916,9 @@ fn check_param_wf(tcx: TyCtxt<'_>, param: &hir::GenericParam<'_>) {
                             )
                             .emit();
                         }
+                        // Should have been normalized in
+                        // `traits::search_for_adt_const_param_violation`
+                        ty::Projection(_) => unreachable!(),
```

Comment:
could be reachable if it can't be normalized due to generics

Reply:
Done!